Improved HFCLK management for nRF5x

Matt-prog opened the pull request:
Improved HFCLK management for nRF5x with weak functions, the user can override and implement it's own HFCLK management, when other peripherials, such as Radio needs it.

**Usage**
User can override those functions:
`void dcd_enable_hfclk(void);` - called, when dcd requests HFCLK activation
`void dcd_disable_hfclk(void);` - called, when dcd requests HFCLK deactivation

If the functions are not overridden, HFCLK will be activated and deactivated by TinyUSB.

Commit:
Improved HFCLK management for nRF5x

pauleffect90 commented:
@Matt-prog Hi matt. Sorry to bother you, but would you be so kind as to tell me what version of the nrf SDK did you use with tinyusb? the latest one seems to be incompatible out of the box.

hathach reviewed (changes requested):
1. Adding new .c and .h are not accepted, please put them all under current dcd file. dcd_* callback name also need changes as well (but that can wait for later). 
2. For this minor changes, please remove your name from the copy right list as well as CONTRIBUTORs.rst. Instead put an comment note at the end of the license term with for your name and the changes.

Matt-prog commented:
Hi @hathach,
thank you for the review! I am currently implementing the suggested changes. I moved the contents of the file `dcd_nrf5x_cb.c` to `dcd_nrf5x.c`.  However, I would keep the header file (or rename it to `dcd_nrf5x.h`). This file can be included in the user .c or .cpp file with the definition of the "strong" functions. For the example, see [here](https://gitlab.nsoric.com/mtf/mcu/nrf52/nrf52840-libraries/propradio/-/blob/HFXO_fix_mf/src/PropRadio.cpp#L24-34) how I implemented it in the library, which controls Radio and HFCLK instead of SoftDevice on nRF52 board.

Do you have any suggestions for callback function names? Does the dcd_ or another prefix should be used?

I have removed my name from the copyright list and the CONTRIBUTORS.rst file. Sorry for the misunderstanding, this is my first pull request on github.
